src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -299,7 +299,7 @@ public new bool IsSectionDefined(string name)
 
             var zone = ThemeLayout.Zones[name];
 
-            return zone != null;
+            return !(zone is ZoneOnDemand);
```

Comment:
why not use `return zone.IsNullOrEmpty()` ?

Reply:
@sebastienros 

Yes, i just created this extension and i plan to use it ;

Reply:
See in my first comment ;)

> Then just created an IShape.IsNullOrEmpty() extension that i will start to use.